src/components/modules/readonly.ts

```diff
@@ -0,0 +1,65 @@
+import Module from '../__module';
+
+/**
+ * @module readonly
+ *
+ * Has one important method:
+ *    - {Function} toggleReadonly - Set read-only mode or toggle current state
+ *
+ * @version 1.0.0
+ *
+ * @typedef {ReadOnly} ReadOnly
+ * @property {boolean} readOnlyEnabled - read-only state
+ */
+export default class ReadOnly extends Module {
+  /**
+   * Value to track read-only state
+   *
+   * @type {boolean}
+   */
+  private readOnlyEnabled = false;
```

Comment:
we can make it public to allow access it from the different methods

Reply:
I implemented `isEnabled()` method that returns its value